src/User.php

```diff
@@ -642,7 +642,7 @@ public function getFromDBbyEmail($email, $condition = [])
     public function getDefaultEmail()
     {
 
-        if (!isset($this->fields['id'])) {
+        if (empty($this->fields['id'])) {
```

Comment:
The `isNewItem()` is the most appropriate for this.

```suggestion
        if ($this->isNewItem()) {
```